src/main/java/me/qoomon/maven/gitversioning/GitVersioningModelProcessor.java

```diff
@@ -399,8 +400,8 @@ private boolean isRelatedPom(File pomFile) throws IOException {
                 && pomFile.getCanonicalPath().startsWith(gitDirectory.getParentFile().getCanonicalPath() + File.separator);
     }
 
-    private static File findGitDir(File baseDirectory) {
-        return new FileRepositoryBuilder().findGitDir(baseDirectory).getGitDir();
+    private static File findGitDir(File baseDirectory) throws IOException{
+        return new FileRepositoryBuilder().setWorkTree(baseDirectory).readEnvironment().findGitDir().build().getDirectory();
```

Comment:
Isn't it the same behaviour?

Reply:
Unfortunately no.
you start from a worktree dir and you want to find the git dir ... 

please make a separate worktree by calling `git worktree add ...`and try it out